src/js/oncoprintlegendrenderer.js

```diff
@@ -154,6 +154,11 @@ var OncoprintLegendView = (function() {
 	var config = rule.getLegendConfig();
 	if (config.type === 'rule') {
 	    var concrete_shapes = rule.apply(config.target, model.getCellWidth(true), view.base_height);
+            if (config.baseGrayRect) {
+                // generate backgrounds
+                root.appendChild(svgfactory.rect(0, 0, model.getCellWidth(true), view.base_height, 'rgb(190,190,190)'));
```

Comment:
sorry but I think I need to ask another change - I don't like how hardcoded this is. What I had originally meant was that the user should be able to specify the base shapes, and I think I need to go back to that - because what if we change the color of the diploid rectangle? Then we'll need to go in and make a change to the oncoprint library each time. That's exactly the use case we were avoiding with the oncoprint refactor - we shouldn't need to mess with the library to modify the visualization rules. Sorry but I think I need to ask you to implement something more general, similar to the generality allowed by legend_config.target

Reply:
Okay, so help me to understand your point. 
Is your plan to have `baseLegendShapes` so it can be configured from `geneticrules.js` ?

I don't understand what do you mean by "generality allowed by legend_config.target" because I don't see in which point user can specify  `legend_config.target`.

Reply:
@adamabeshouse 

Thank you for your suggestions.

While I can appreciate making things more general, we cannot keep making changes. We have already overspent the budget on this feature by a fair margin.

Would it be possible to merge this as is now, and reporting the requested improvement as an issue? Then we can finalize this part, and work on the issue in time that is spent outside of a project, for instance in between projects. This would allow us to move on with our client.

Thanks for understanding.

Reply:
@Sjoerd-van-Hagen I understand your concerns, but I have to say I don’t think over-budget a justifiable reason to merge code. I am afraid if we merge it and log the issue, we would never go back to fix it. I know budgeting is hard, but what we have to make sure is that the code we merge is good. Sorry about that.

We decided to make the oncoprint library general and we tried really hard to achieve that. I don't think we should break that for this feature.

@adamabeshouse could you explain more about your comments / address @rnugraha's question?

Reply:
@rnugraha a cheaper option that I would feel comfortable with is to get rid of the rule legend_config.baseGrayRect member, and instead add a member to the genetic rule set parameters, `baseLegendRectColor`, that, if set, determines that the base rect will be drawn, at the given color. That way we could specify the color from outside oncoprintjs.

What do you think?

Reply:
@jjgao I agree that over-budget is not a reason to merge code and the code must be of sufficient quality. On the other hand we cannot keep making improvements and polishing it indefinitely. Perhaps it would be good if Riza and Adam can schedule a call to go over this if it is unclear what needs to be done.

@adamabeshouse thanks for considering other options, much appreciated!